cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -1652,6 +1660,158 @@ runTransactionCalculateMinValueCmd
+              return (currentEra, systemStart, eraHistory, utxo)
+        )
+        & onLeft (left . TxCmdQueryConvenienceError . AcqFailure)
+        & onLeft (left . TxCmdQueryConvenienceError)
+
+    Refl <-
+      testEquality nodeEra (convert txEra)
+        & hoistMaybe
+          ( TxCmdTxSubmitErrorEraMismatch $
+              EraMismatch{ledgerEraName = docToText $ pretty nodeEra, otherEraName = docToText $ pretty txEra}
+          )
+
+    pparams <- getProtocolParams sbe localNodeConnInfo
+
+    carryTransactionCalculatePlutusScriptCostCmd
+      (convert txEra)
+      systemStart
+      eraHistory
+      pparams
+      txEraUtxo
+      txBody
+   where
+    getProtocolParams
+      :: ShelleyBasedEra era -> LocalNodeConnectInfo -> ExceptT TxCmdError IO (LedgerProtocolParameters era)
+    getProtocolParams sbe localNodeConnInfo = do
+      let qInMode = QueryInEra $ QueryInShelleyBasedEra sbe Api.QueryProtocolParameters
+      pp <-
+        executeQueryAnyMode localNodeConnInfo qInMode
+          & modifyError TxCmdQueryConvenienceError
+      return $ LedgerProtocolParameters pp
+
+    carryTransactionCalculatePlutusScriptCostCmd
+      :: CardanoEra era
+      -> SystemStart
+      -> EraHistory
+      -> LedgerProtocolParameters era
+      -> UTxO era
+      -> TxBody era
+      -> ExceptT TxCmdError IO ()
+    carryTransactionCalculatePlutusScriptCostCmd era' systemStart eraHistory pparams txEraUtxo txBody = do
+      scriptHashes <-
+        monoidForEraInEon @AlonzoEraOnwards era' (\aeo -> pure $ collectScriptHashes aeo txBody txEraUtxo)
+          & hoistMaybe (TxCmdAlonzoEraOnwardsRequired era')
+
+      executionUnitPrices <-
+        pure (getExecutionUnitPrices era' pparams) & onNothing (left TxCmdPParamExecutionUnitsNotAvailable)
+
+      scriptExecUnitsMap <-
+        firstExceptT (TxCmdTxExecUnitsErr . AnyTxCmdTxExecUnitsErr) $
+          hoistEither $
+            evaluateTransactionExecutionUnits
+              era'
+              systemStart
+              (toLedgerEpochInfo eraHistory)
+              pparams
+              txEraUtxo
+              txBody
+
+      scriptCostOutput <-
+        firstExceptT TxCmdPlutusScriptCostErr $
+          hoistEither $
+            renderScriptCostsWithScriptHashes
+              executionUnitPrices
+              scriptHashes
+              scriptExecUnitsMap
+      liftIO $ LBS.writeFile (unFile outputFile) $ encodePretty scriptCostOutput
+     where
+      collectScriptHashes
+        :: AlonzoEraOnwards era
+        -> TxBody era
+        -> UTxO era
+        -> Map
+            ScriptWitnessIndex
+            ScriptHash
+      collectScriptHashes aeo tb utxo =
+        alonzoEraOnwardsConstraints aeo $
+          let ShelleyTx _ ledgerTx' = makeSignedTransaction [] tb
+              ledgerUTxO = toLedgerUTxO (convert aeo) utxo
+           in getPurpouses aeo $ L.getScriptsNeeded ledgerUTxO (ledgerTx' ^. L.bodyTxL)
+       where
```

Comment:
Please do not nest `where` clauses like this.

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/1031/commits/4fd80b359c0b20b6ac49d61a89f8185982bea251